src/ImageSharp/Formats/Bmp/BmpDecoder.cs

```diff
@@ -43,9 +43,7 @@ public Image<TPixel> Decode<TPixel>(Configuration configuration, Stream stream)
             {
                 Size dims = decoder.Dimensions;
 
-                // TODO: use InvalidImageContentException here, if we decide to define it
-                // https://github.com/SixLabors/ImageSharp/issues/1110
-                throw new ImageFormatException($"Can not decode image. Failed to allocate buffers for possibly degenerate dimensions: {dims.Width}x{dims.Height}. This error can happen for very large RLE bitmaps, which are not supported.", ex);
+                throw new InvalidImageContentException($"Can not decode image. Failed to allocate buffers for possibly degenerate dimensions: {dims.Width}x{dims.Height}. This error can happen for very large RLE bitmaps, which are not supported.", ex);
```

Comment:
Can you move these and others to the `ThrowHelper` classes for each format please. That's why I missed them.